File: online-store-api/online-store-api/src/main/java/com/code81/onlinestore/service/AddressService.java
package com.code81.onlinestore.service;

import com.code81.onlinestore.dto.address.AddressRequest;
import com.code81.onlinestore.dto.address.AddressResponse;

import java.util.List;

public interface AddressService {

    List<AddressResponse> list(Long customerId);

    AddressResponse add(Long customerId, AddressRequest request);

    AddressResponse update(Long customerId, Long addressId, AddressRequest request);

    void delete(Long customerId, Long addressId);
}

File: code81_online_store_angular/src/app/core/services/address.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { AddressRequest, AddressResponse } from '../models/address.model';

@Injectable({ providedIn: 'root' })
export class AddressService {
  private http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiUrl}/customers/me/addresses`;

  list(): Observable<AddressResponse[]> {
    return this.http.get<AddressResponse[]>(this.baseUrl);
  }

  add(request: AddressRequest): Observable<AddressResponse> {
    return this.http.post<AddressResponse>(this.baseUrl, request);
  }

  update(id: number, request: AddressRequest): Observable<AddressResponse> {
    return this.http.put<AddressResponse>(`${this.baseUrl}/${id}`, request);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
